### PR TITLE
Integration tests, fixed compile warning in websockets test, fixed se…

### PR DESCRIPTION
**Pull request by JonathanHenson**
…gfault in elasticurl.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/test_websocket_decoder.c [hunk 1]
```diff
@@ -185,7 +185,8 @@ DECODER_TEST_CASE(websocket_decoder_rsv) {
         };
 
         /* Set the appropriate RSV */
-        input[0] |= (uint8_t)(1 << (6 - rsv));
+        /* the bit arithmetic is setup this way to avoid Conversion warnings from the compiler. */
+        input[0] = (uint8_t)(input[0] | (1 << (6 - rsv)));
```
Comment (graebm) [inline tests/test_websocket_decoder.c]: I wonder if it's a better idea to just disable the compiler warnings in places than rewrite bit-twiddling code so it's even harder to understand. Bit-twiddling code is hard enough to grok as it is.

File: integration-testing/elasticurl_test.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (graebm) [inline integration-testing/elasticurl_test.sh]: Is it worth writing this in Python so we don't need a separate script for Windows?

File: integration-testing/elasticurl_test.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+set -e
+set -x
+
+echo "elasticurl binary is located at $1"
+
+$1 -v TRACE example.com
+$1 -v TRACE -P -H "content-type: application/json" -i -d "{'test':'testval'}" http://httpbin.org/post
+$1 -v TRACE -o elastigirl.png https://s3.amazonaws.com/code-sharing-aws-crt/elastigirl.png
```
Comment (graebm) [inline integration-testing/elasticurl_test.sh]: comments pls

`# Sanity check, see if request/response complete without error`

I'm not really sure what the 2nd cmd is testing

`# Download a file with elasticurl and curl, ensure the results are identical`

File: tests/test_websocket_decoder.c (hunk 1 above)
Comment (JonathanHenson) [reply tests/test_websocket_decoder.c]: I strongly dislike the wConversion option as worthless due to its false positive rate. If you all want to remove the warning fine. But it doesn’t build as is.

File: integration-testing/elasticurl_test.sh (hunk 3 above)
Comment (JonathanHenson) [reply integration-testing/elasticurl_test.sh]: 2nd one actually sends body data

**Pull request merged**